ambry-api/src/main/java/com/github/ambry/account/AccountService.java

```diff
@@ -107,6 +108,36 @@
    */
   public boolean removeAccountUpdateConsumer(Consumer<Collection<Account>> accountUpdateConsumer);
 
+  /**
+   * Add a new container to an existing account.
+   * @param accountName the name of account in which new container will be added.
+   * @param newContainer the new container to add.
+   * @return a pair of account id and container id strings associated with existing account and new container.
+   */
+  default Pair<String, String> addContainer(String accountName, Container newContainer) {
+    throw new UnsupportedOperationException("This method is not supported");
+  }
+
+  /**
+   * Update attributes of an existing container.
+   * @param accountName the name of account which container belongs to.
+   * @param updatedContainer the updated {@link Container} used to replace current one.
+   * @return {@code true} if container is successfully updated. {@code false} otherwise.
+   */
+  default boolean updateContainer(String accountName, Container updatedContainer) {
```

Comment:
How about calling this updateContainers (plural) and accepting a collection of containers?

Reply:
Yeah, I actually renamed it already, will push out the change to github after rest changes are done.